core-java-modules/core-java-io-3/src/test/java/com/baeldung/lastmodifiedfile/LastModifiedFileAppUnitTest.java

```diff
@@ -0,0 +1,69 @@
+package com.baeldung.lastmodifiedfile;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class LastModifiedFileAppUnitTest {
+
+    private final static String SOURCEDIRECTORY = "src/test/resources/lastmodfiles";
+
+    @BeforeAll
+    public static void setUpFiles() throws IOException {
+        File srcDir = new File(SOURCEDIRECTORY);
+        if (!srcDir.exists())
+            srcDir.mkdir();
+
+        FileUtils.cleanDirectory(srcDir);
+
+    }
+
+    @Test
+    public void givenDirectory_whenUsingIoApi_thenFindLastModfile() throws IOException {
+        Path file01 = Paths.get(SOURCEDIRECTORY + "/file01.txt");
```

Comment:
@azhwani I know I am caviling here, but the first test is testing against 1 file only, so we don't really test anything, the first modified file would also pass :)

Reply:
@doodoroma, I have made some changes, can you please take a look ;)

Reply:
It was not clear, but the situation is true for all tests, in fact. Unit tests must be independent from each other. Maybe a small fixture could be added?